web/src/dialog_widget.ts

```diff
@@ -203,17 +190,17 @@ export function submit_api_request(
     request_method({
         url,
         data,
-        success(response_data?: string) {
+        success(response_data, textStatus, jqXHR) {
             close_modal();
             if (success_continuation !== undefined) {
-                success_continuation(response_data);
+                success_continuation(response_data, textStatus, jqXHR);
             }
         },
-        error(xhr?: JQuery.jqXHR) {
+        error(xhr, error_type, xhn) {
             ui_report.error(failure_msg_html, xhr, $("#dialog_error"));
             hide_dialog_spinner();
             if (error_continuation !== undefined) {
-                error_continuation(xhr);
+                error_continuation(xhr, error_type, xhn);
```

Comment:
Have you checked that the actual `error_continuation` and `success_continuation` values won't behave differently if passed these extra parameters? It's probably fine, in that I expect most of them don't even look at their parameters at all, but we should check.

Reply:
@timabbott  Yup, I have cross-checked all the calls made to `submit_api_request`, and as you said not even a single one of them looks at these parameters at all, mostly these callbacks are being used to run some logic after the request has finished.
So, I think it would be no harm in passing these extra parameters in case we need them in the future.